content/docs/guides/parallel-execution.md

```diff
@@ -0,0 +1,396 @@
+
+- Add the Maven **Surefire plugin configuration** to the `build` section of the `POM`.
+
+```shell
+<plugin>
+	<groupId>org.apache.maven.plugins</groupId>
+	<artifactId>maven-surefire-plugin</artifactId>
+	<version>2.22.0</version>
+</plugin>
+```
+
+- Use the Maven `install` or a suitable command to **execute the POM**. This should run in parallel thread mode. You should see a result similar to below. The **scenarios and rows of the scenario outlines are executed in different threads**.
+
+```shell
+Thread ID - 15 - Scenario Outline Row 2 from scenariooutlines feature file.
+Thread ID - 14 - Scenario Outline Row 1 from scenariooutlines feature file.
+Thread ID - 16 - Scenario 1 from scenarios feature file.
+Thread ID - 17 - [Scenario 2 from scenarios feature file.
+```
+
+- To execute using a Maven **Failsafe plugin**, setup the `POM` as described in the JUnit section. Remove the `parallel` and `useUnlimitedThreads` settings in the `configuration` part.
+
+
+The default **thread count of the dataprovider** in parallel mode is **10**. To change this the `dataproviderthreadcount` property needs to be added to the `configuration` section of the Surefire or Failsafe plugin in the `POM`.
+```shell
+<configuration>
+	<properties>
+    	<property>
+        	<name>dataproviderthreadcount</name>
+        	<value>20</value>
+    	</property>
+	</properties>
+</configuration>
+```
+
+If you have **multiple runners**, set the parallel configuration of `classes` to reduce execution times. In addition the `threadCount` can be set to to the desired value or `useUnlimitedThreads` can be set to true.
+```shell
+<configuration>
+	<parallel>classes</parallel>
+	<threadCount>4</threadCount>
+</configuration>
+```
+
+For a **visual representation** you can add the **timeline report** with the plugin option to a `CucumberOptions` annotation in the runner class. Scroll to the end for an image of the report.
+
+
+# CLI
+The `Main class` in the `cucumber.api.cli package` is used to execute the feature files. There is no need to create any runner class. The usage options for this class is mentioned [here](https://github.com/cucumber/cucumber-jvm/blob/v4.0.0/core/src/main/resources/cucumber/api/cli/USAGE.txt). The `--threads` option needs to be set to a value **greater than 1** to run in parallel. When the parallel mode is used, the scenarios and rows in a scenario outline will be run in multiple threads.
+ 
+Below is the basic command to start the execution.
+```shell
+java -cp <classpath> cucumber.api.cli.Main -g <steps package> --threads <thread count> <path to feature files>
+```
+
+Follow the below steps to **execute the command from a terminal**.
+
+- Download the necessary cucumber jars from the cucumber.io Maven repository (links provided under version number) or, if available, copy from the local repository into a folder. These need to be provided in the **classpath**. The versions mentioned are compatible with the latest cucumber release 4.3.0.

```

Comment:
Please, let's **not** describe downloading the jars, as we really recommend not doing that. I realise you are doing this here to give an alternative to getting the dependencies through Maven, but doing so leads to many problems and also I'd rather not maintain this list in the documentation.
In fact, in that case it might be better to also leave out the sections about the Maven dependencies for JUnit and TestNG, and just link to the part of the docs where that is already described. Which will save some duplication also :)

Reply:
Removed the dependencies in CLI section.
For the other two the 10 minute tutorial uses the cucumber-archetype. Is that updated to get latest versions? I tried it out around 3 months back and it added some pretty old version,

Reply:
The archetype has been updated to a more recent version (v4.?) but probably not *the* most recent version... It's a great thing to use to get started, but a pain to maintain.

Reply:
Using the archetype is a good solution. As long as it gets a V4 it works, unless in a future version parallel functionality is removed. Will add the link.